src/tests/stream/index.test.ts

```diff
@@ -185,4 +182,110 @@ describe("Stream", () => {
+
+        factory("4mb => ReadMB", mb(4), { controlCharacter: ControlCharacters.ReadMB, number: 0, remainder: 0 });
+        factory("5mb => ReadMB", mb(5), { controlCharacter: ControlCharacters.ReadMB, number: 0, remainder: mb(1) });
+        factory("8mb => ReadMB", mb(8), { controlCharacter: ControlCharacters.ReadMB, number: 1, remainder: 0 });
+        factory("2000mb yields number=255", mb(2000), { controlCharacter: ControlCharacters.ReadMB, number: 255, remainder: mb(2000 - 256) });
+    });
+
+    describe("encode", () => {
+        function factory(name: string, input: Buffer, output: Buffer): void {
+            it(name, () => {
+                expect(stream.encode(input)).toEqual(output);
+            });
+        }
+
+        factory("nothing => returns empty buffer", Buffer.alloc(0), Buffer.alloc(0));
+        factory("one byte => read-byte once", Buffer.from([0x00]), Buffer.from([ControlCharacters.ReadByte, 0x00]));
+        factory(
+            "two bytes => read-byte twice",
+            Buffer.from([0x00, 0x01]),
+            Buffer.from([ControlCharacters.ReadByte, 0x00, ControlCharacters.ReadByte, 0x01])
+        );
+        factory(
+            "three bytes => read-byte three times",
+            Buffer.from([0x00, 0x01, 0x02]),
+            Buffer.from([ControlCharacters.ReadByte, 0x00, ControlCharacters.ReadByte, 0x01, ControlCharacters.ReadByte, 0x02])
+        );
+        factory(
+            "four bytes => read-bytes once",
+            Buffer.from([0x00, 0x01, 0x02, 0x03]),
+            Buffer.from([ControlCharacters.ReadBytes, 0x00, 0x00, 0x01, 0x02, 0x03])
+        );
+        factory(
+            "five bytes => read-bytes once and a read-byte",
+            Buffer.from([0x00, 0x01, 0x02, 0x03, 0x04]),
+            Buffer.from([ControlCharacters.ReadBytes, 0x00, 0x00, 0x01, 0x02, 0x03, ControlCharacters.ReadByte, 0x04])
+        );
+        factory(
+            "eight bytes => read-bytes 2x",
+            Buffer.from([0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07]),
+            Buffer.from([ControlCharacters.ReadBytes, 0x01, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07])
+        );
+        factory(
+            "15 bytes does what you'd expect",
+            Buffer.alloc(15).fill(0xff),
+            Buffer.from([
+                ControlCharacters.ReadBytes,
+                0x02,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                0xff,
+                ControlCharacters.ReadByte,
+                0xff,
+                ControlCharacters.ReadByte,
+                0xff,
+                ControlCharacters.ReadByte,
+                0xff
+            ])
+        );
+        // TODO: more?
```

Comment:
maybe test error cases; run a coverage report to find out

Reply:
@ei-pi [Coverage Report](https://app.codecov.io/gh/thetayloredman/node-subdata-2/pull/1)